MarcusBarnes/mik
# Issue #441 - Change metadata manipulators in OAI toolchains to manipulate entire XML document

**Github issue**: (#441)

# What does this Pull Request do?

Modifies metadata parsers used in OAI-PMH toolchains to use metadata manipulators that have access to the entire metadata XML file.

# What's new?

* Modifies the DC and MODS OAI-PMH metadata parsers to use metadata manipulators that have access to the entire serialized XML metadata document instead of Cdm and Csv-style XML snippets.
* Provides a new metadata manipulator, SimpleReplaceOai.

# How should this be tested?

The changes in this PR do not have PHPUnit tests, they must be smoketested.

* Check out the issue-441 branch.
* Run `composer dump-autoload` or the eqivalent on your system.
* Using the attached .ini file, run mik: `./mik -c issue-441.ini`
  * The DC files in the output directory should say "Super Special" in their dc.rights elements.
* Edit the .ini file so that you are harvesting MODS instead of DC (uncomment the `metadata_prefix = mods` and `class = mods\OaiToMods` lines).
* Delete your output and temp directories and rerun MIK.
  * The MODS files in the output directory should say "Super Special" in their `<accessCondition>` elements.  

# Additional Notes

* A wiki entry documenting the new metadata maniipulator is available at https://github.com/MarcusBarnes/mik/wiki/Metadata-manipulator:-SimpleReplaceOai. It should be linked from https://github.com/MarcusBarnes/mik/wiki/Manipulators and all of the OAI-PMH toolchain wiki pages.

# Interested parties

@MarcusBarnes @bondjimbond 

[issue-441.ini.txt](https://github.com/MarcusBarnes/mik/files/1394909/issue-441.ini.txt)
